src/components/modal-container.js

```diff
@@ -154,15 +154,17 @@ export default function ModalContainerFactory(
       const {selectedDataset, dataType, filtered} = uiState.exportData;
       // get the selected data
       const filename = 'kepler-gl';
-      const selectedDatasets = datasets[selectedDataset] ? [datasets[selectedDataset]] : Object.values(datasets);
+      const selectedDatasets = datasets[selectedDataset] ?
+        [datasets[selectedDataset]] : Object.values(datasets);
+
       if (!selectedDatasets.length) {
         // error: selected dataset not found.
         this._closeModal();
       }
 
       selectedDatasets.forEach(selectedData => {
-        const {allData, data, fields, label} = selectedData;
-        const exportData = filtered ? data : allData;
+        const {allData, fields, label, filteredIdxCPU = []} = selectedData;
+        const exportData = filtered && filteredIdxCPU ? filteredIdxCPU.map(i => allData[i]) : allData;
```

Comment:
since you have a default value for filteredIdxCPU in the previous line, do you need to perform a check filteredIdxCPU?
maybe you get a null value rather than undefined?